tensorflow/compiler/mlir/xla/tests/hlo-legalize-to-lhlo.mlir

```diff
@@ -11,39 +11,39 @@ func @attrs_copy(%operand: memref<2x2xf32>, %result: memref<2x2xf32>) {
   return
 }
 
-// CHECK-LABEL: func @func_op
-func @func_op(%arg0: tensor<4xf32>, %arg1: tensor<4xf32>) -> tensor<4xf32> {
-  // CHECK: (%[[NEW_ARG0:.*]]: memref<4xf32>, %[[NEW_ARG1:.*]]: memref<4xf32>, %[[RESULT:.*]]: memref<4xf32>)
-  %0 = xla_hlo.max %arg0, %arg1 {name = "maximum.47"} : tensor<4xf32>
-  // CHECK-NEXT: "xla_lhlo.max"(%[[NEW_ARG0]], %[[NEW_ARG1]], %[[RESULT]])
-  return %0 : tensor<4xf32>
-  // CHECK-NEXT: "xla_lhlo.terminator"() : () -> ()
-}
+// CHECK-LABEL: func @buffer_assignment_on_func_with_branches_using_live_analysis
+func @buffer_assignment_on_func_with_branches_using_live_analysis(%arg0: tensor<10xf32>, %arg1: tensor<11x11xf32>) -> tensor<10xf32> {
+  br ^bb1(%arg0, %arg1: tensor<10xf32>, tensor<11x11xf32>)
+^bb1(%arg2: tensor<10xf32>, %arg3: tensor<11x11xf32>):
+  %0 = "xla_hlo.exp"(%arg2) : (tensor<10xf32>) -> tensor<10xf32>
+  br ^bb2(%0: tensor<10xf32>)
+^bb2(%arg4: tensor<10xf32>):
+  %2 = "xla_hlo.exp"(%arg4) : (tensor<10xf32>) -> tensor<10xf32>
+  return %2 : tensor<10xf32>
+}
+//      CHECK: (%[[ARG0:.*]]: memref<10xf32>, %[[ARG1:.*]]: memref<11x11xf32>, %[[RESULT:.*]]: memref<10xf32>)
+// CHECK-NEXT: %[[BB1_ARG0:.*]] = alloc() {temp = true} : memref<10xf32>
+// CHECK-NEXT: "xla_lhlo.copy"(%[[ARG0]], %[[BB1_ARG0]]) :
+// CHECK-NEXT: %[[BB1_ARG1:.*]] = alloc() {temp = true} : memref<11x11xf32>
+// CHECK-NEXT: "xla_lhlo.copy"(%[[ARG1]], %[[BB1_ARG1]])
+// CHECK-NEXT: br ^bb1(%[[BB1_ARG0]], %[[BB1_ARG1]] : memref<10xf32>, memref<11x11xf32>)
+// CHECK-NEXT: ^bb1(%[[ARG2:.*]]: memref<10xf32>, %[[ARG3:.*]]: memref<11x11xf32>): // pred: ^bb0
+// CHECK-NEXT: %[[EXP_RESULT:.*]] = alloc() {temp = true} : memref<10xf32>
+// CHECK-NEXT: dealloc %[[ARG3]] : memref<11x11xf32>
```

Comment:
i think i am missing smth: can `arg3` dealloc be done before `exp_result` alloc?